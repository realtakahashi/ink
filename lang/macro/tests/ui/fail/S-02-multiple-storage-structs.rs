use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod multiple_storage_structs {
    #[ink(storage)]
    struct FirstStorageStruct {}

    // ink! currently doesn't allow for multiple #[ink(storage)] structs
    #[ink(storage)]
    struct SecondStorageStruct {}

    impl FirstStorageStruct {
        #[ink(constructor)]
        fn constructor1() -> Self {
            Self {}
        }

        #[ink(message)]
        fn message1(&self) {}
    }

    impl SecondStorageStruct {
        #[ink(constructor)]
        fn constructor2() -> Self {
            Self {}
        }

        #[ink(message)]
        fn message2(&self) {}
    }
}

fn main() {}
