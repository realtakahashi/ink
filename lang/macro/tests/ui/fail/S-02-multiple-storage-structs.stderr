error: encountered 2 conflicting storage structs
 --> $DIR/S-02-multiple-storage-structs.rs:3:1
  |
3 | #[ink::contract(version = "0.1.0")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: conflicting storage struct
 --> $DIR/S-02-multiple-storage-structs.rs:6:12
  |
6 |     struct FirstStorageStruct {}
  |            ^^^^^^^^^^^^^^^^^^

error: conflicting storage struct
  --> $DIR/S-02-multiple-storage-structs.rs:10:12
   |
10 |     struct SecondStorageStruct {}
   |            ^^^^^^^^^^^^^^^^^^^
